### PR TITLE
Jk/cumulus 3258 main

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses portion of [CUMULUS-3258](https://bugs.earthdata.nasa.gov/browse/CUMULUS-3258)

This PR updates the current terraform provider to v5 to address downstream dependency issues/upgrade Core dependencies. 

This PR updates Core integrations to use/test Orca v8.1.0. 

## Changes

- **CUMULUS-3258**
  - Updated all hashicorp terraform AWS provider configs to ~> 5.0.0
    - Upstream/downstream terraform modules will need to utilize an AWS provider
      that matches this range
- **CUMULUS-3258**
  - Update @cumulus/api/lib/orca/getOrcaRecoveryStatusByGranuleCollection
    to o/@cumulus/api/lib/orca/getOrcaRecoveryStatusByGranuleIdAndCollection and
    add collectionId to arguments to support Orca v8+ required us of
    collectionId
  - Update all Core integration tests/integrations to be compatible with Orca >=
    v8.1.0 only
  - Updated all terraform AWS providers to ~> 5.0.0
- **CUMULUS-3258**
  - Fix un-prefixed s3 lifecycle configuration ID from CUMULUS-2915
 

## PR Checklist

- [x] Update CHANGELOG
- [ ] Unit tests
- [x] Ad-hoc testing - Deploy changes and test manually
- [x] Integration tests

### CODE REVIEW

**Comment by Jkovarik** [comment]
This PR is based on work tested in https://github.com/nasa/cumulus/pull/3429 by disabling ORCA dependency.   This PR cannot merge until work is done to update ORCA to work with the v5 AWS TF provider.

File: example/cumulus-tf/ingest_and_publish_granule_with_orca_workflow.tf [hunk 1]
```diff
@@ -19,7 +19,6 @@ module "ingest_and_publish_granule_with_orca_workflow" {
       hyrax_metadata_updates_task_arn: module.cumulus.hyrax_metadata_updates_task.task_arn,
       lzards_backup_task_arn: module.cumulus.lzards_backup_task.task_arn,
       post_to_cmr_task_arn: module.cumulus.post_to_cmr_task.task_arn,
-      orca_copy_to_archive_adapter_task: module.cumulus.orca_copy_to_archive_adapter_task.task_arn
```
Comment (npauzenga) [inline example/cumulus-tf/ingest_and_publish_granule_with_orca_workflow.tf:22]: Why is this removed?

File: example/cumulus-tf/main.tf [hunk 2]
```diff
@@ -139,9 +139,7 @@ module "cumulus" {
   oauth_provider   = var.oauth_provider
   oauth_user_group = var.oauth_user_group
 
-  orca_api_uri = module.orca[0].orca_api_deployment_invoke_url
-  orca_lambda_copy_to_archive_arn = module.orca[0].orca_lambda_copy_to_archive_arn
-  orca_sfn_recovery_workflow_arn  = module.orca[0].orca_sfn_recovery_workflow_arn
+  # orca_api_uri = module.orca[0].orca_api_deployment_invoke_url
```
Comment (npauzenga) [inline example/cumulus-tf/main.tf:142]: Same question as above. What're these changes?

File: packages/db/src/config.ts [hunk 3]
```diff
@@ -38,6 +38,7 @@ export const getSecretConnectionConfig = async (
     password: dbAccessMeta.password,
     database: dbAccessMeta.database,
     port: dbAccessMeta.port ?? 5432,
+    ssl: true,
```
Comment (npauzenga) [inline packages/db/src/config.ts:41]: Presumably necessary due to the updated provider?

**Review by npauzenga — COMMENTED** [review]
Just a quick question on a couple lines. Otherwise this looks good pending integration tests and Orca's changes.

File: example/cumulus-tf/ingest_and_publish_granule_with_orca_workflow.tf (hunk 1 above)
Comment (Jkovarik) [reply example/cumulus-tf/ingest_and_publish_granule_with_orca_workflow.tf]: Removing as outdated

File: example/cumulus-tf/main.tf (hunk 2 above)
Comment (Jkovarik) [reply example/cumulus-tf/main.tf]: Removing as outdated

File: tf-modules/archive/s3_bucket_lifecycle_config.tf [hunk 4]
```diff
@@ -1,16 +1,16 @@
 data "aws_s3_bucket" "system_bucket" {
-  bucket = var.system_bucket 
+  bucket = var.system_bucket
 }
 
 resource "aws_s3_bucket_lifecycle_configuration" "system_bucket_lifecycle_config" {
   bucket = data.aws_s3_bucket.system_bucket.id
   rule {
-    id     = "expire_temporary_execution_status_files"
+    id     = "${var.prefix}_expire_temporary_execution_status_files"
```
Comment (Jkovarik) [inline tf-modules/archive/s3_bucket_lifecycle_config.tf:8]: This is a fix for a bug introduced in https://github.com/nasa/cumulus/pull/3123/files

File: packages/db/src/config.ts (hunk 3 above)
Comment (Jkovarik) [reply packages/db/src/config.ts:41]: This got dragged into the v5 commit, but that doesn't make sense - this shouldn't matter for our internal code.... 

Let me look into it.

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,20 +10,36 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+  - Added clarity to how the bucket field needs to be configured for the
+    move-granules task definition
 
 ### Changed
 
 - Security upgrade node from 14.19.3-buster to 14.21.1-buster
+- **CUMULUS-3258**
+  - Update @cumulus/api/lib/orca/getOrcaRecoveryStatusByGranuleCollection
+    to o/@cumulus/api/lib/orca/getOrcaRecoveryStatusByGranuleIdAndCollection and
+    add collectionId to arguments to support Orca v8+ required us of
```
Comment (npauzenga) [inline CHANGELOG.md:38]:
```suggestion
    add collectionId to arguments to support Orca v8+ required use of
```

File: example/cumulus-tf/main.tf [hunk 6]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 3.14.1"
+      version = "~> 5.0"
```
Comment (npauzenga) [inline example/cumulus-tf/main.tf:5]: Is there a reason to constrain it to v5.x? It would prevent breakages if e.g. v6 requires changes but maybe that's ok and preferable to manual updates?

File: example/cumulus-tf/orca.tf [hunk 7]
```diff
@@ -12,8 +12,8 @@ locals {
 
 # ORCA Module
 module "orca" {
-  count = 1
-  source = "https://github.com/nasa/cumulus-orca/releases/download/v8.0.1/cumulus-orca-terraform.zip"
+  aws_region = var.region
```
Comment (npauzenga) [inline example/cumulus-tf/orca.tf:15]: Is the `count` param here removed because the orca module doesn't support it? Obviously we'd only want 1 Orca instance so at best it's redundant I'd guess.

File: example/cumulus-tf/vpc.tf [hunk 8]
```diff
@@ -5,10 +5,12 @@ data "aws_vpc" "application_vpc" {
   }
 }
 
-data "aws_subnet_ids" "subnet_ids" {
+data "aws_subnets" "subnet_ids" {
   count = length(var.lambda_subnet_ids) == 0 ? 1 : 0
-  vpc_id = var.vpc_id != null ? var.vpc_id : data.aws_vpc.application_vpc[0].id
-
+  filter {
```
Comment (npauzenga) [inline example/cumulus-tf/vpc.tf:10]: What does `filter` do in this context? From what I've seen it's supposed to "narrow down" results but here we have a block that just assigns values...?

File: example/spec/parallel/orca/OrcaBackupAndRecoverySpec.js [hunk 9]
```diff
@@ -149,8 +149,9 @@ describe('The S3 Ingest Granules workflow', () => {
 
   describe('the recovery workflow', () => {
     let asyncOperationId;
+    let collectionId;
 
-    it('generates an async operation through the Cumulus API', async () => {
+    it('generatess an async operation through the Cumulus API', async () => {
```
Comment (npauzenga) [inline example/spec/parallel/orca/OrcaBackupAndRecoverySpec.js:154]:
```suggestion
    it('generates an async operation through the Cumulus API', async () => {
```

File: example/spec/parallel/orca/OrcaBackupAndRecoverySpec.js [hunk 10]
```diff
@@ -162,11 +163,12 @@ describe('The S3 Ingest Granules workflow', () => {
       }
       const recoveryWorkflowName = get(collectionsApiResponse, 'meta.granuleRecoveryWorkflow');
 
+      collectionId = encodedConstructCollectionId(collection.name, collection.version);
```
Comment (npauzenga) [inline example/spec/parallel/orca/OrcaBackupAndRecoverySpec.js:166]: This method was introduced to URL encode the collectionId. If we're using the collectionId in a body, not a path, I think we could probably use `constructCollectionId` but also, it really doesn't matter.

**Review by npauzenga — CHANGES_REQUESTED** [review]
Looks good @Jkovarik! Just a couple questions/comments.

File: example/cumulus-tf/orca.tf (hunk 7 above)
Comment (Jkovarik) [reply example/cumulus-tf/orca.tf:15]: Correct - Orca removed that parameter.

File: example/cumulus-tf/vpc.tf (hunk 8 above)
Comment (Jkovarik) [reply example/cumulus-tf/vpc.tf:10]: So - the object type has changed here from aws_subnet_ids to aws_subnets (https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/subnet.html).

Unlike aws_subnet_ids (which uses vpc_id to return , the aws_subnets data source returns *all* subnets for an account - this is not what we want as we *only* only subnets in the VPC Core is set to work in.    Bear in mind we're using this only for defaults in the example project where var.lambda_subnet_ids is *not* set (it will instead grab a single subnet at random-ish from a vpc subnet *or* the application_vpc if they also do not have a vpc configured. 

So, to answer the question, in this context, filter is filtering out a single aws subnet that's in the VPC Core is in, or the first application_vpc if for some reason that's not configured.

File: example/cumulus-tf/main.tf (hunk 6 above)
Comment (npauzenga) [reply example/cumulus-tf/main.tf:5]: Yeah, understood on the risks. Just wondering if that approach generally is right. We can discuss outside of this PR.

**Review by npauzenga — APPROVED** [review]
Looks good to me @Jkovarik! Thanks for addressing the feedback!
